### ISSUE TITLE
Alpha Channels in PSDs (Part 3)

### ISSUE THREAD

**Issue by blue-j** [ImageMagick/ImageMagick#4549]
### ImageMagick version

7.1.0-16

### Operating system

MacOS

### Operating system, version and so on

11.6.1

### Description

@dlemstra You're code to support extracting multiple alpha channels with PSD files works great!  We cannot use the "-channel all" switch, but if we just leave it out entirely, it produces all the channels.

`magick input.psd -separate output_channel%d.jpg`

Is this all as intended?

So, Photoshop format files also support background transparency, as one would normally expect with alpha channels.  For some reason, Adobe chooses not to show it in the interface in the Channels palette, but they do store the transparency as the first alpha channel after the color channels.

However, when we use the same command above (including with/out  "`-define psd:alpha-unblend=off`") we get not only all the channels, but duplicates of the color channels and the Transparency channel!

![alpha_cha0](https://user-images.githubusercontent.com/3990142/144768083-807b80db-d3cc-4c4c-b130-38475f9f9306.jpg)
![alpha_cha12](https://user-images.githubusercontent.com/3990142/144768085-5f7e744a-fe34-4047-b9f4-6f0307cd1cb7.jpg)
![alpha_cha11](https://user-images.githubusercontent.com/3990142/144768086-3f6929b0-ed07-4833-87e6-b8f3af949df6.jpg)
![alpha_cha10](https://user-images.githubusercontent.com/3990142/144768087-da11002e-9ed3-421f-bb58-03ae633fe7a3.jpg)
![alpha_cha9](https://user-images.githubusercontent.com/3990142/144768088-7cfeca4a-6f86-4bf7-9f1b-bee27a14a7a5.jpg)
![alpha_cha8](https://user-images.githubusercontent.com/3990142/144768090-e546f928-1f14-44c7-bef2-8313a415cca5.jpg)
![alpha_cha7](https://user-images.githubusercontent.com/3990142/144768091-77b166bb-ace8-48c6-be71-e575b29d60dc.jpg)
![alpha_cha6](https://user-images.githubusercontent.com/3990142/144768092-5dd9ee7d-0151-4cca-bcf1-282ad83a9ebc.jpg)
![alpha_cha5](https://user-images.githubusercontent.com/3990142/144768093-9335aea2-799d-4efa-9cb1-562501962efa.jpg)
![alpha_cha4](https://user-images.githubusercontent.com/3990142/144768096-d0d07c2c-e8ad-492d-b9e5-465c88f06cb2.jpg)
![alpha_cha3](https://user-images.githubusercontent.com/3990142/144768097-3a93b355-afc7-49aa-af2c-6e0f7dc34b0d.jpg)
![alpha_cha2](https://user-images.githubusercontent.com/3990142/144768099-ac32ffb8-2da9-45b9-b079-d91446e0ce8a.jpg)
![alpha_cha1](https://user-images.githubusercontent.com/3990142/144768100-9030e68c-6ed0-472c-9204-b31ca3fe4dcf.jpg)

### Steps to Reproduce

simply run this command on the PSD provided:

`magick alpha_test_5_WITH_BG_TRANSPARENCY.psd -separate output_channel%d.jpg`

(We have tried with "-channel all" and with  "-define psd:alpha-unblend=off" and that produces no difference in output)

### Images

[alpha_test_5_WITH_BG_TRANSPARENCY.psd.zip](https://github.com/ImageMagick/ImageMagick/files/7656993/alpha_test_5_WITH_BG_TRANSPARENCY.psd.zip)

**Comment by dlemstra** [comment]
What is the expected result can you show this visually? I am having trouble understanding what your issue is.

**Comment by blue-j** [comment]
> What is the expected result can you show this visually? I am having trouble understanding what your issue is.

If you run the command on the sample, you'll see that the all channels are output 2x instead of only once.  trying "-channel all" doesn't produce any difference.

**Comment by blue-j** [comment]
<img width="464" alt="Screen Shot 2022-01-18 at 2 35 07 PM" src="https://user-images.githubusercontent.com/3990142/150029647-2e925c90-3d33-4d19-86cd-579636c83bff.png">

**Comment by dlemstra** [comment]
Can you give it another try @blue-j?
